src/domain/interfaces.py

```
@@ -0,0 +1,122 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional, AsyncIterator, Dict, Any, Tuple
```

Ruff (0.12.2)

Import from `collections.abc` instead: `AsyncIterator`

Import from `collections.abc`

(UP035)